sdk/java/pulumi/src/main/java/com/pulumi/test/Mocks.java

```diff
@@ -0,0 +1,163 @@
+package com.pulumi.test;
+
+import com.google.common.collect.ImmutableMap;
+
+import javax.annotation.Nullable;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Hooks to mock the engine that provide test doubles for offline unit testing of stacks.
+ */
+public interface Mocks {
+
+    /**
+     * Invoked when a new resource is created by the program.
+     * <p>
+     *
+     * @param args arguments containing resource information
+     * @return A resource identifier and resource state. State can be either a POJO
+     * or a dictionary bag. The returned ID may be null for component resources.
+     */
+    CompletableFuture<ResourceResult> newResourceAsync(ResourceArgs args);
+
+    /**
+     * Invoked when the program needs to call a provider to load data (e.g., to retrieve an existing resource).
+     * <p>
+     * Default implementation returns an empty map.
+     *
+     * @param args arguments containing call information
+     * @return the call invocation result, can be either a POCO or a dictionary bag.
+     */
+    default CompletableFuture<Map<String, Object>> callAsync(CallArgs args) {
+        return CompletableFuture.completedFuture(Map.of());
+    }
+
+    /**
+     * ResourceArgs for use in {@link #newResourceAsync(ResourceArgs)}
+     */
+    class ResourceArgs {
```

Comment:
moved from a separate file, this way it does not need to be imported and can have a shorter name